components/client/src/test/unit/java/com/hotels/styx/client/connectionpool/SimpleConnectionPoolTest.java

```diff
@@ -921,4 +922,98 @@ public void ensureInEstablishmentCountIsDecrementedInError() throws InterruptedE
                 .verifyComplete();
     }
 
+    @Test
+    public void emitsExceptionWhenBrrowingFromClosedPool() {
+
+        SimpleConnectionPool pool = new SimpleConnectionPool(origin, defaultConnectionPoolSettings(), connectionFactory);
+        pool.close();
+        StepVerifier.create(pool.borrowConnection())
+                .expectError(IllegalStateException.class)
+                .verify();
+        assertEquals(pool.stats().connectionAttempts(), 0);
+        assertEquals(pool.stats().busyConnectionCount(), 0);
+        assertEquals(pool.stats().availableConnectionCount(), 0);
+        assertEquals(pool.stats().pendingConnectionCount(), 0);
+    }
+
+    @Test
+    public void justClosesConnectionWhenReturningToClosedPool() {
+        when(connectionFactory.createConnection(any(Origin.class), any(ConnectionSettings.class))).thenReturn(Mono.just(connection1));
+        SimpleConnectionPool pool = new SimpleConnectionPool(origin, defaultConnectionPoolSettings(), connectionFactory);
+
+        StepVerifier.create(pool.borrowConnection())
+                .expectNext(connection1)
+                .verifyComplete();
+        assertEquals(pool.stats().connectionAttempts(), 1);
+        assertEquals(pool.stats().busyConnectionCount(), 1);
+
+        pool.close();
+        assertEquals(0, pool.stats(). closedConnections());
+
+        doAnswer( arg -> {
+            when(connection1.isConnected()).thenReturn(false);
+            return true;
+        }).when(connection1).close();
+        pool.returnConnection(connection1);
+        assertEquals(pool.stats().connectionAttempts(), 1);
+        assertEquals(pool.stats().busyConnectionCount(), 0);
+        assertEquals(1, pool.stats(). closedConnections());
+    }
+
+    @Test
+    public void closesConnectionAndUpdatesMetricsWhenClosingConnectionFromClosedPool() {
```

Comment:
This test is to show that a non-active pool doesn't substitute a closed TCP connection. Therefore I'd consider either renaming this test to something like `closedPoolDoesNotReplaceClosedTcpConnection`.

Or perhaps even better clarify this in a comment?

Reply:
Let's try with a clearer name.